### ISSUE TITLE
Wrong behavior when Layout/ClosingParenthesisIndentation sees double splat

### ISSUE THREAD

**Issue by bquorning** [rubocop/rubocop#11483]
Consider the following code in test.rb:
```rb
foo(bar,
  **baz
)
```

## Expected behavior

`./exe/rubocop --only=Layout/ClosingParenthesisIndentation test.rb` should not see any offenses. [The cop documentation](https://github.com/rubocop/rubocop/blob/6139e4019cfa349fbddaf7244eeb21e780a1a6d4/lib/rubocop/cop/layout/closing_parenthesis_indentation.rb#L63-L68) says “when other params are not lined up on multiple lines, outdent right paren by `IndentationWidth`”. Also this code does not cause offense:

```rb
foo(bar,
  baz
)
```

## Actual behavior

```
Offenses:

test.rb:3:1: C: [Correctable] Layout/ClosingParenthesisIndentation: Align ) with (.
)
^
```

## Steps to reproduce the problem

See above.

## RuboCop version

```
❯ ./exe/rubocop -V
1.44.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.2) [x86_64-darwin21]
  - rubocop-performance 1.15.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.18.1
```

```
❯ git rev-parse @                                                                      
51ded37abcd63590cca74a86e2b26812e5ac8ae3
```
